### ISSUE TITLE
Custom sink example does not compile

### ISSUE THREAD

**Issue by csparker247** [gabime/spdlog#1259]
I'm on macOS Mojave (10.14.1) and spdlog 1.4.1 is installed via Homebrew. Using the documentation [here](https://github.com/gabime/spdlog/wiki/4.-Sinks#implementing-your-own-sink), I'm trying to create a custom sink. I have not yet done any modification to the example code. However, this results in the following error during compilation:
```Shell
../core/src/Logging.cpp:27:9: error: use of undeclared identifier 'sink'
        sink::formatter_->format(msg, formatted);
        ^
1 error generated.
```

I can get it to find `formatter_` if I instead reference from `base_sink<Mutex>`:
```C++
fmt::memory_buffer formatted;
spdlog::sinks::base_sink<Mutex>::formatter_->format(msg, formatted);
```
However, the compiler then complains that `formatted` cannot be bound to the second parameter:
```Shell
../core/src/Logging.cpp:25:66: error: non-const lvalue reference to type 'basic_memory_buffer<[...], 250>' cannot bind to a value of unrelated type 'basic_memory_buffer<[...], (default) inline_buffer_size aka 500>'
        spdlog::sinks::base_sink<Mutex>::formatter_->format(msg, formatted);
                                                                 ^~~~~~~~~
```
I can only successfully compile if `formatted` matches the `memory_buf_t` typedef:
```C++
fmt::basic_memory_buffer<char, 250> formatted;
spdlog::sinks::base_sink<Mutex>::formatter_->format(msg, formatted);
```

**Comment by csparker247** [comment]
UPDATE: I see that a fix for this [might be in v1.4.2](https://github.com/gabime/spdlog/commit/f5492aed12b144df73faf8ae336b3dbc39ebbc38#diff-074bb8c76cb5c003cf19a1efc18d217e). Will wait for this release to propagate to Homebrew.

**Comment by gabime** [comment]
The example in the wiki is outdated unfortunately.
The correct way would be (as you already figured out):
```c++
spdlog::memory_buf_t formatted;
base_sink<Mutex>::formatter_->format(msg, formatted);
```
